### ISSUE TITLE
Unexpected behaviours on "Support of OSGi Bundle and Versioning annotations" feature

### ISSUE THREAD

**Issue by niport** [eclipse-pde/eclipse.pde#696]
The “Support of OSGi Bundle and Versioning annotations” feature was first introduced in 2022-09 (see https://eclipse.dev/eclipse/news/4.25/pde.php), but I think it presents some issues.

I did the checks below in 2022-09 and 2023-06 on Windows 10.

## Removal of a previously added Bundle annotation

If certain annotations are added and then removed, the `MANIFEST.MF` file is not updated accordingly.

For instance it happens with the `@Export` and the `@Header` annotations. If an `@Export` annotation is added in the` package-info.java` file the `MANIFEST.MF` file is updated accordingly, but if then it is removed the `MANIFEST.MF` still contain the package in the `Export-Package` bundle header.

For instance add and then remove the following lines or even remove the whole `package-info.java` file

```
@org.osgi.annotation.bundle.Export
@org.osgi.annotation.bundle.Header(name="Bundle-Category", value="osgi")
@org.osgi.annotation.bundle.Header(name="My-Extra-Header", value="Set by annotations")
```

## Use of java constants in Bundle annotations

The following annotation does not work as expected (this example is explicitly proposed in the OSGi spec at §10.8.8):

`@org.osgi.annotation.bundle.Header(name=org.osgi.framework.Constants.BUNDLE_CATEGORY, value="osgi")`

but the following works:

`@org.osgi.annotation.bundle.Header(name=”Bundle-Category”, value="osgi")`

In both cases to avoid compilation errors I need to add explicitly  import the `org.osgi.framework` package. I don’t think that this `Import-Package` is correct because this dependency is not required at runtime.

Fix proposal are not available on `@Export` annotation

If the simple annotation name is used instead of a fully qualified name, Eclipse doesn’t behave in the same way in the suggestion to quick-fix the issue:

 `@Export` annotation

![image](https://github.com/eclipse-pde/eclipse.pde/assets/36211095/f5737478-1bad-4b96-952f-232bc624c884)

`@Header` annotation (1)
 
![image](https://github.com/eclipse-pde/eclipse.pde/assets/36211095/e72f8199-97cb-49fd-9e45-68f35b32c856)

`@Header` annotation (2)

![image](https://github.com/eclipse-pde/eclipse.pde/assets/36211095/22a00c17-e607-4df0-8d72-18a4b1754d0b)

Eventually, after having added the necessary import in the package-info.file, the following lines compile correctly (remember that I need to add explicitly org.osgi.framework in the `MANIFEST.MF` imported packages, but as I stated above the third `@Header` annotation doesn't produce any entry in the `MANIFEST.MF` file.

```
@Export
@Header(name="Bundle-Category", value="osgi")
@Header(name="My-Extra-Header", value="Set by annotations")
@Header(name=Constants.BUNDLE_CATEGORY, value="osgi")
package org.acme.annotations;

import org.osgi.annotation.bundle.Export;
import org.osgi.annotation.bundle.Header;
import org.osgi.framework.Constants;
```

## @Requirement and @Capability bundle annotations are not supported.
 
If one the above bundle annotations in added in the source code the `MANIFEST.MF` corresponding headers are not generated.

Moreover the following example (from the OSGi specifications) doesn’t compile since the `org.osgi.namespace.extender` plugin is not present in the `Plug-in Dependences` section of the project:

```
@Requirement(namespace=ExtenderNamespace.EXTENDER_NAMESPACE,
  name="osgi.component", version="1.4.0")
```

**Comment by laeubi** [comment]
@niport yes the annotation support is rather basic her, do you like to give it a try an improve some of the issue you have seen?

For the issue with
> Fix proposal are not available on @Export annotation
> If the simple annotation name is used instead of a fully qualified name, Eclipse doesn’t behave in the same way in the suggestion to quick-fix the issue:

I think this is a [JDT issue](https://github.com/eclipse-jdt/eclipse.jdt.core/issues) that seem not to find the annotation in its full form?

**Comment by niport** [comment]
> @niport yes the annotation support is rather basic her, do you like to give it a try an improve some of the issue you have seen?
> 
> For the issue with
> 
> > Fix proposal are not available on @export annotation
> > If the simple annotation name is used instead of a fully qualified name, Eclipse doesn’t behave in the same way in the suggestion to quick-fix the issue:
> 
> I think this is a [JDT issue](https://github.com/eclipse-jdt/eclipse.jdt.core/issues) that seem not to find the annotation in its full form?

I am not a JDT and PDE internals expert but it is possible. I will file an issue in JDT. Thanks!

**Comment by niport** [comment]
I eventually gave up (for the moment) in submitting the issue on JDT because the issue is a little bit more complex to reproduce.

**Comment by niport** [comment]
Sorry I forgot to answer you about the contribute. No, I really don't think I could have time. Also because PDE is very complex and the startup time to be able to be productive is long. In any case I found the recent improvments very valuable.
